url/src/host.rs

```diff
@@ -269,7 +269,7 @@ fn ends_in_a_number(input: &str) -> bool {
     } else {
         last
     };
-    if !last.is_empty() && last.chars().all(|c| ('0'..='9').contains(&c)) {
+    if !last.is_empty() && last.chars().all(|c| c.is_ascii_digit()) {
```

Comment:
You can replace `chars()` with `as_bytes()` here and lower, as it expects for ASCII range only, which is probably faster, than walking over utf8 code points.

Reply:
Ah yes, very good point!